### DISCUSSION TITLE
swift-syntax identity conflicts

### DISCUSSION THREAD

**Discussion by michalnowak061** [pointfreeco/swift-snapshot-testing#924] [General]
'swift-snapshot-testing' dependency on 'https://github.com/swiftlang/swift-syntax' conflicts with dependency on 'https://github.com/apple/swift-syntax.git' which has the same identity 'swift-syntax'. this will be escalated to an error in future versions of SwiftPM.

**Expected behavior**
zero warnings

**Screenshots**
<img width="1434" alt="Screenshot 2024-11-01 at 22 59 55" src="https://github.com/user-attachments/assets/c298f023-e6d9-4e79-b879-6244e845d06a">

**Environment**
  - swift-snapshot-testing version 1.17.6
  - Xcode 16.1
  - Swift 6.0.0

**Comment by mbrandonw** [comment]
Hi @michalnowak061, we [depend](https://github.com/pointfreeco/swift-snapshot-testing/blob/42a086182681cf661f5c47c9b7dc3931de18c6d7/Package.swift#L24) on the most current `swiftlang/swift-syntax` repo, so if you are getting this warning it must mean that you are either using an old version of the library, or you have another dependency (possibly transitively) on `apple/swift-syntax`.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Reply by michalnowak061** [reply]
Good point. I actually found the old url in a different package. Interesting that SPM shows an error in your library, even though it contains the latest url
